packages/suite/src/hooks/wallet/useCoinmarketP2pOffers.ts

```diff
@@ -0,0 +1,118 @@
+import { createContext, useCallback, useContext, useEffect, useState } from 'react';
+import { ContextValues, P2pStep, UseOffersProps } from '@wallet-types/coinmarketP2pOffers';
+import { useActions, useDevice, useSelector } from '@suite-hooks';
+import { useTimer } from '@trezor/react-utils';
+import { InvityAPIReloadQuotesAfterSeconds } from '@wallet-constants/coinmarket/metadata';
+import * as coinmarketP2pActions from '@wallet-actions/coinmarketP2pActions';
+import invityAPI from '@suite-services/invityAPI';
+import { P2pQuote } from 'invity-api';
+import { useCoinmarketNavigation } from '@wallet-hooks/useCoinmarketNavigation';
+import * as coinmarketCommonActions from '@wallet-actions/coinmarket/coinmarketCommonActions';
+
+export const useOffers = ({ selectedAccount }: UseOffersProps): ContextValues => {
+    const timer = useTimer();
+
+    const { account } = selectedAccount;
+    const { isLocked } = useDevice();
+    const [selectedQuote, setSelectedQuote] = useState<P2pQuote>();
+    const { navigateToP2pForm } = useCoinmarketNavigation(account);
+    const { openCoinmarketP2pConfirmModal, submitRequestForm } = useActions({
+        openCoinmarketP2pConfirmModal: coinmarketP2pActions.openCoinmarketP2pConfirmModal,
+        submitRequestForm: coinmarketCommonActions.submitRequestForm,
+    });
+    const { device, providers, quotesRequest, quotes } = useSelector(state => ({
+        device: state.suite.device,
+        providers: state.wallet.coinmarket.p2p.p2pInfo?.providers,
+        quotesRequest: state.wallet.coinmarket.p2p.quotesRequest,
+        quotes: state.wallet.coinmarket.p2p.quotes,
+    }));
+    const [innerQuotes, setInnerQuotes] = useState(quotes);
+    const [providerVisited, setProviderVisited] = useState(false);
+    const [p2pStep, setP2pStep] = useState(P2pStep.GET_STARTED);
+
+    const getQuotes = useCallback(async () => {
+        if (!selectedQuote && quotesRequest) {
+            timer.loading();
+            const response = await invityAPI.getP2pQuotes(quotesRequest);
+
+            if (!response || response.quotes?.length === 0) {
+                timer.stop();
+                return;
+            }
+
+            setInnerQuotes(response.quotes);
+            timer.reset();
+        }
+    }, [selectedQuote, quotesRequest, timer]);
+
+    useEffect(() => {
+        if (!quotesRequest) {
+            navigateToP2pForm();
+            return;
+        }
+
+        if (!timer.isLoading && !timer.isStopped) {
+            if (timer.resetCount >= 40) {
+                timer.stop();
+            }
+
+            if (timer.timeSpend.seconds === InvityAPIReloadQuotesAfterSeconds) {
+                getQuotes();
+            }
+        }
+    });
```

Comment:
maybe missing dependency array?

Reply:
Is it? My IDE is not complaining in this case.

Reply:
IDE doesn't complain if you don't define dependency array at all. It just means, that the effect is processed after every render. Maybe it's just a nitpick.

Reply:
Well, my IDE complains e.g. in case of missing values when calling `useCallback`. Anyway, `useEffect` is called the same way for other trade types, it's sort of copy&paste code so I'll leave it like this.